lib/Mongo/MongoCollection.php

```diff
@@ -96,6 +110,19 @@ public function __get($name)
         return $this->db->selectCollection($this->name . '.' . $name);
     }
 
+    public function __set($name, $value)
+    {
+        if ('w' == $name) {
+            $this->w = $value;
+            $this->setWriteConcern($this->w, $this->wtimeout);
+        } elseif ('wtimeout' == $name) {
+            $this->wtimeout = $value;
+            $this->setWriteConcern($this->w, $this->wtimeout);
+        } else {
+            trigger_error('trying to set invalid property', E_ERROR);
```

Comment:
While it isn't really a smart usecase, currently it's possible to set any property on `MongoCollection` object and read it again, so this error should not be thrown. I'd say we simply set the property, you can check the `MongoId` class for an implementation.